src/coreclr/jit/liveness.cpp

```diff
@@ -2182,6 +2182,9 @@ bool Compiler::fgTryRemoveNonLocal(GenTree* node, LIR::Range* blockRange)
 //
 void Compiler::fgRemoveDeadStoreLIR(GenTree* store, BasicBlock* block)
 {
+    if (!PreciseRefCountsRequired())
+        return;
+
```

Comment:
Ideally we should not be depending on dead store elimination for correctness (it is also a bit weird - ref counts are only tangentially related to liveness and are really for CSE and register allocator to use). Fine to defer though.
```suggestion
    // TODO-LLVM: we should not need this for correctness, investigate removing.
    if (!PreciseRefCountsRequired())
        return;

```